chain-impl-mockchain/src/certificate/mod.rs

```diff
@@ -64,6 +64,49 @@ impl<'a> CertificateSlice<'a> {
     }
 }
 
+#[derive(Clone)]
+pub enum CertificatePayload {
+    StakeDelegation(PayloadData<StakeDelegation>),
+    OwnerStakeDelegation(PayloadData<OwnerStakeDelegation>),
+    PoolRegistration(PayloadData<PoolRegistration>),
+    PoolRetirement(PayloadData<PoolRetirement>),
+    PoolUpdate(PayloadData<PoolUpdate>),
+}
+
+impl<'a> From<&'a Certificate> for CertificatePayload {
+    fn from(certificate: &'a Certificate) -> Self {
+        match certificate {
+            Certificate::StakeDelegation(payload) => {
+                CertificatePayload::StakeDelegation(payload.payload_data())
+            }
+            Certificate::OwnerStakeDelegation(payload) => {
+                CertificatePayload::OwnerStakeDelegation(payload.payload_data())
+            }
+            Certificate::PoolRegistration(payload) => {
+                CertificatePayload::PoolRegistration(payload.payload_data())
+            }
+            Certificate::PoolRetirement(payload) => {
+                CertificatePayload::PoolRetirement(payload.payload_data())
+            }
+            Certificate::PoolUpdate(payload) => {
+                CertificatePayload::PoolUpdate(payload.payload_data())
+            }
+        }
+    }
+}
+
+impl<'a> From<&'a CertificatePayload> for CertificateSlice<'a> {
```

Comment:
I don't think it's a good idea to call "owned-data-to-borrow-slice" through `From`. I'ld much rather have a specific method (unless there's one generic trait that allow this, like `Borrow` which I don't think exist here due to <'a>)